waltz-tools/src/main/java/com/wepay/waltz/tools/client/ClientCli.java

```diff
@@ -273,12 +170,14 @@ private void consumeAndValidate(WaltzClientConfig config) throws Exception {
             private int txnPerThread;
             private int avgInterval;
             private WaltzClient client;
+            private CountDownLatch producerTransactionsCompleted;
```

Comment:
Looks like you are using `producerTransactionsCompleted` and `allProducerDone` to determine when the producers have successfully submitted the transactions. 
I feel like we just make use of one CountDownLatch right? 
You can have a `allProducerTxnsSubmitted` latch defined just like the other CountDownLatch and you can use it in the HighWaterMarkTxnContext.onCompletion(..) and also in the `ProducerThread` and in `produceTransactions`.
What do you think? That way you don't have to pass it as an argument to the `HighWaterMarkTxnContext`.